test/application_system_test_case.rb

```diff
@@ -34,11 +34,17 @@ def assert_active(selector, error_msg = nil, wait: nil)
     assert_equal element, page.active_element, "Expected #{selector} to be the active element, but it is not. #{error_msg}"
```

Comment:
Everything from here downward is: (a) adding new tests, (b) addressing some flappy test cases, and (c) creating some new test helpers to simplify a bunch of testing of scroll position